tulip/transys/transys.py

```diff
@@ -928,53 +1229,77 @@ def _output_fts(ts, transitions, sol):
     AP = ts.aps
     ts_simu.atomic_propositions.add_from(AP)
     for i in range(n_cells):
-        ts_simu.states.add(i, ap=ts.nodes[next(iter(sol[i]))]['ap'])
+        ts_simu.states.add(
+            i,
+            ap=ts.nodes[next(iter(sol[i]))]['ap'])
     for i in range(n_cells):
         for j in range(n_cells):
             if transitions[j, i]:
                 ts_simu.transitions.add(i, j)
     return ts_simu, Part_hash
 
 
-def simu_abstract(ts, simu_type):
-    """Create a bi/dual-simulation abstraction for a Finite Transition System.
-
-    @param ts: input finite transition system, the one you want to get
-                    its bi/dual-simulation abstraction.
-    @type ts: L{FTS}
-    @param simu_type: string 'bi'/'dual', flag used to switch b.w.
-                      bisimulation algorithm and dual-simulation algorithm.
-    @return: the bi/dual simulation, and the corresponding partition.
-    @rtype: L{FTS}, C{dict}
+def simu_abstract(
+        ts:
+            FTS,
+        simu_type:
+            _ty.Literal[
+                'bi',
+                'dual']
+        ) -> tuple[
+            FTS,
+            dict]:
+    """Abstract `ts`, using bi/dual-simulation.
+
+    Create a bi/dual-simulation abstraction for
+    the finite transition system `ts`.
+
+    @param ts:
+        input finite transition system,
+        the one you want to get
+        its bi/dual-simulation abstraction.
+    @param simu_type:
+        string 'bi'/'dual', flag used to switch b.w.
```

Comment:
```suggestion
        string 'bi'/'dual', flag used to switch between
```
Not the fault of this PR, but the abbreviation "b.w." for "between" is not common, so I recommend that it is expanded.